pkg/pillar/types/assignableadapters.go

```diff
@@ -460,6 +524,30 @@ func (aa *AssignableAdapters) LookupIoBundleIfName(ifname string) *IoBundle {
 	return nil
 }
 
+type ownParentError struct{}
+
+func (o ownParentError) Error() string {
+	return "IOBundle cannot be it's own parent"
+}
+
```

Comment:
Is this approach the recommended golang approach to define error strings?
I think the old/original standard packages just define a well-known string for the exported errors.

Reply:
In the standard library I see for example:
```
var EOF = errors.New("EOF") 
```

for most of the newly introduced errors, this approach would work (I guess), but not for the collision error, because the error string is different depending with what it is colliding.

```
var me3 = errors.New("myErr3")
var me4 = errors.New("myErr4")

errors.Is(me3, me3)
```
evaluates to `false`, because it is evaluating the members.

---

An alternative approach would be to use `errors.Is` and have global instances of `ownParentError`, `parentAssigngrpMismatchError`, `emptyAssigngrpWithParentError` and `cycleDetectedError` and then create
the `ioBundleCollisionErr` with a
```
func (i ioBundleCollisisonErr) Is(target err) bool {
    return reflect.TypeOf(i) == reflect.TypeOf(m)
}
```
But are those the same error if they are different collisions?

I have to think about it a bit more.

Reply:
> But are those the same error if they are different collisions?
> 
> I have to think about it a bit more.

But the Is() implementation can consider the fields in the error struct as well...

Reply:
> But the Is() implementation can consider the fields in the error struct as well...

I think that is the default implementation anyways.


I am now using `errors.New` and according to the documentation:
> func New(text string) error
    New returns an error that formats as the given text. Each call to New
    returns a distinct error value even if the text is identical.


so that works and the collisionError still uses different error strings, but because I check with `reflect.TypeOf` they are detected as the same.